Dockerfile

```diff
@@ -112,7 +102,8 @@ RUN --mount=type=cache,target=/root/.cache/pip \
         && export SCCACHE_IDLE_TIMEOUT=0 \
         && export CMAKE_BUILD_TYPE=Release \
         && sccache --show-stats \
-        && python3 setup.py bdist_wheel --dist-dir=dist --py-limited-api=cp38 \
+        && PIP_EXTRA_INDEX_URL="$([ "${TARGETPLATFORM}" = "linux/arm64" ] && echo "https://download.pytorch.org/whl/nightly/cu124")" \
```

Comment:
is there a way we could abstract the URL? it seems like we'd have to update at least a couple of places when it changes.

Reply:
Good point, I found it odd already that we had to specify the extra index url, I moved all of this to `requirements-cuda.txt`.

cc @youkaichao